pep-0696.rst

```diff
@@ -0,0 +1,470 @@
+that hasn't been overridden it should be treated like it was
+substituted into the ``TypeAlias``. However, it can be specialised
+further down the line.
+
+.. code:: py
+
+   class SomethingWithNoDefaults(Generic[T, T2]): ...
+
+   MyAlias: TypeAlias = SomethingWithNoDefaults[int, DefaultStrT]  # valid
+   reveal_type(MyAlias())        # type is SomethingWithNoDefaults[int, str]
+   reveal_type(MyAlias[bool]())  # type is SomethingWithNoDefaults[int, bool]
+
+   MyAlias[bool, int]  # Invalid: too many arguments passed to MyAlias
+
+Subclassing
+'''''''''''
+
+Subclasses of ``Generic``\ s with ``TypeVarLike``\ s that have defaults
+behave similarly to ``Generic`` ``TypeAlias``\ es.
+
+.. code:: py
+
+   class SubclassMe(Generic[T, DefaultStrT]): ...
+
+   class Bar(SubclassMe[int, DefaultStrT]): ...
+   reveal_type(Bar())        # type is Bar[str]
+   reveal_type(Bar[bool]())  # type is Bar[bool]
+
+   class Foo(SubclassMe[int]): ...
+
+   reveal_type(Foo())  # type is <subclass of SubclassMe[int, int]>
+
+   Foo[str]  # Invalid: Foo cannot be further subscripted
+
+   class Baz(Generic[DefaultIntT, DefaultStrT]): ...
+
+   class Spam(Baz): ...
+   reveal_type(Spam())  # type is <subclass of Baz[int, str]>
+
+Using bound and default
+'''''''''''''''''''''''
+
+If both ``bound`` and ``default`` are passed ``default`` must be a
+subtype of ``bound``. Otherwise the type checker should generate an
+error.
+
+.. code:: py
+
+   TypeVar("Ok", bound=float, default=int)     # Valid
+   TypeVar("Invalid", bound=str, default=int)  # Invalid: the bound and default are incompatible
+
+Constraints
+'''''''''''
+
+For constrained ``TypeVar``\ s, the default needs to be one of the
+constraints. A type checker should generate an error even if it is a
+subtype of one of the constraints.
+
+.. code:: py
+
+   TypeVar("Ok", float, str, default=float)     # Valid
+   TypeVar("Invalid", float, str, default=int)  # Invalid: expected one of float or str got int
+
+Function Defaults
+'''''''''''''''''
+
+``TypeVarLike``\ s currently can only be used where a parameter can go unsolved.
+
+.. code:: py
+
+   def foo(a: DefaultStrT | None = None) -> DefaultStrT: ...
+
+   reveal_type(foo(1)) # type is int
+   reveal_type(foo())  # type is str
+
+If they are used where the parameter type is known, the defaults
+should just be ignored and a type checker can emit a warning.
+
+
+Implementation
+--------------
+
+At runtime, this would involve the following changes to the ``typing``
+module.
+
+- a new class ``_DefaultMixin`` needs to be added which is a superclass
+  of ``TypeVar``, ``ParamSpec``, and ``TypeVarTuple``.
+
+   -  the type passed to default would be available as a ``__default__``
+      attribute.
+
+The following changes would be required to both ``GenericAlias``\ es:
+
+-  logic to determine the defaults required for a subscription.
+
+   -  potentially a way construct ``types.GenericAliases`` using a
```

Comment:
Again, please remove this. It's not necessary and it's orthogonal to this PEP's major purpose.